### PR TITLE
Fix infinite error loop during pagination

### LINKED ISSUE #1: Infinite error loop during pagination
Receive a 529 Service Unavailable error when following pagination links using purest client. Version number is included in request url twice (i.e. https://api.constantcontact.com/v2//v2/emailmarketing/campaigns?next=cGFnZVNpemU9NCZwYWdlTnVtPTU).

### PR DESCRIPTION
**Pull request by szb0018**
Resolves #1

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -36,21 +36,40 @@ function ConstantContact(config = {}) {
     dotProp.set(this, path, resource);
   }
 
-  const follow = (url) => Resource.invokeRequest(this.client().get(url));
-
-  this.paginate = (first, resolve, reject, finished) => {
-    first.catch(reject).then((result) => {
-      resolve(result);
+  const follow = (url) => {
+    let version = url.match(/^\/(v\d+)\//)[1];
```
Comment (compwright) [inline lib/index.js:40]: Is `url` a full URL? Perhaps we could eliminate the URL parsing and just go with `url` rather than `this.client().get(url)`.

Comment (szb0018) [reply lib/index.js:40]: this is what gets returned `"next_link": "/v2/emailmarketing/campaigns?next=cGFnZVNpemU9NCZwYWdlTnVtPTU"`

The Client is configured for paths relative to https://api.constantcontact.com/v2/ while the API returns paths relative to https://api.constantcontact.com/. The result a url with 2x "/v2" paths that is incorrect. IMHO the Client approach is the correct way; API's should only be concerned with paths relative to their root resource - domains and versioning are outside their scope.

Comment (szb0018) [reply lib/index.js:40]: Plus I need to the request to go through the client to append the authentication tokens and track the requests for rate-limiting. So the only options are to change the one url or to remove the version from the client.

Comment (compwright) [reply lib/index.js:40]: Thanks, that helps.

**Pull request merged**